File: src/main/java/io/bindernews/thegrackle/cards/FireWithin.java
package io.bindernews.thegrackle.cards;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.core.AbstractCreature;
import io.bindernews.bnsts.CardVariables;
import io.bindernews.thegrackle.power.FireWithinPower;

public class FireWithin extends BaseCard {
    public static final CardConfig C =
            new CardConfig("FireWithin", CardType.POWER, CardRarity.UNCOMMON, CardTarget.SELF);
    static final CardVariables VARS = CardVariables.config(c -> {
        c.cost(1);
        c.magic(2, 3);
    });

    public FireWithin() {
        super(C, VARS);
    }

    @Override
    public void apply(AbstractCreature p, AbstractCreature m) {
        addToBot(new ApplyPowerAction(p, p, new FireWithinPower(p, magicNumber)));
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/cards/FireWithin.kt
package io.bindernews.thegrackle.cards

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction
import com.megacrit.cardcrawl.core.AbstractCreature
import io.bindernews.bnsts.CardVariables
import io.bindernews.thegrackle.power.FireWithinPower

class FireWithin : BaseCard(C, VARS) {
    override fun apply(p: AbstractCreature, m: AbstractCreature?) {
        addToBot(ApplyPowerAction(p, p, FireWithinPower(p, magicNumber)))
    }

    companion object {
        @JvmField val C = CardConfig("FireWithin", CardType.POWER, CardRarity.UNCOMMON, CardTarget.SELF)
        val VARS = CardVariables.config { c ->
            c.cost(1)
            c.magic(2, 3)
        }
    }
}